pkg/util/cmd/split_args_test.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import "testing"
+
+func TestReturnsAnArrayOfStrings(t *testing.T) {
+	args := SplitCmdString(`date 1423`)
+
+	if args[0] != "date" {
+		t.Errorf("Expected the first arg to be 'date', but was '%s'", args[0])
+	}
+
+	if args[1] != "1423" {
+		t.Errorf("Expected the second arg to be '1423', but was '%s'", args[1])
+	}
+}
+
+func TestRemovesOuterSingleQuotes(t *testing.T) {
+	args := SplitCmdString(`date '1423'`)
+
+	if args[0] != "date" {
+		t.Errorf("Expected the first arg to be 'date', but was '%s'", args[0])
+	}
+
+	if args[1] != "1423" {
+		t.Errorf("Expected the second arg to be '1423', but was '%s'", args[1])
+	}
+}
+
+func TestRemovesOuterDoubleQuotes(t *testing.T) {
+	args := SplitCmdString(`date "1423"`)
+
+	if args[0] != "date" {
+		t.Errorf("Expected the first arg to be 'date', but was '%s'", args[0])
+	}
+
+	if args[1] != "1423" {
+		t.Errorf("Expected the second arg to be '1423', but was '%s'", args[1])
+	}
+}
+
+func TestProperlyHandlesWhitespaceWithinSingleQuotes(t *testing.T) {
+	args := SplitCmdString(`date -f '%a %b %d %T %Z %Y' "01/01/1900" '+%s'`)
```

Comment:
If we use approach suggested here https://github.com/minishift/minishift/pull/764/files/14f24adb92d3ed3f95fc78c4c2f0dbee7d5b7d2f#r112129096, then I believe, this test case would be much smaller.